lib/whitehall_importer.rb

```diff
@@ -1,27 +1,42 @@
 # frozen_string_literal: true
 
 module WhitehallImporter
-  def self.import(whitehall_document)
-    record = WhitehallImport.create!(
+  def self.import_and_sync(whitehall_document)
+    whitehall_import = WhitehallImport.create!(
       whitehall_document_id: whitehall_document["id"],
       payload: whitehall_document,
       content_id: whitehall_document["content_id"],
       state: "importing",
     )
 
-    begin
-      Import.call(whitehall_document)
-      record.update!(state: "completed")
-    rescue StandardError => e
-      record.update!(error_log: e.message,
-                     state: "failed")
-    end
+    import(whitehall_import)
 
-    record
+    sync(whitehall_import) if whitehall_import.imported?
+
+    whitehall_import
+  end
+
+  def self.import(whitehall_import)
+    raise "Cannot import with a state of #{whitehall_import.state}" unless whitehall_import.importing?
```

Comment:
I don't think this exception should be caught by this method (should be a begin / end) - as we don't to change the state of the import on this sort of error.

Same for sync.

Reply:
Good point!